src/main/java/com/google/codeu/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,74 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.codeu.data.Marker;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /** Responds with a JSON array containing marker data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    List<Marker> markers = getMarkers();
+    Gson gson = new Gson();
+    String json = gson.toJson(markers);
+
+    response.getOutputStream().println(json);
+  }
+
+  /** Accepts a POST request containing a new marker. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    double lat = Double.parseDouble(request.getParameter("lat"));
+    double lng = Double.parseDouble(request.getParameter("lng"));
+    String content = Jsoup.clean(request.getParameter("content"), Whitelist.none());
+
+    Marker marker = new Marker(lat, lng, content);
+    storeMarker(marker);
+  }
+
+  /** Fetches markers from Datastore. */
+  private List<Marker> getMarkers() {
+    List<Marker> markers = new ArrayList<>();
+
+    Query query = new Query("Marker");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String content = (String) entity.getProperty("content");
+
+      Marker marker = new Marker(lat, lng, content);
+      markers.add(marker);
```

Comment:
Maybe you could make the two variable names a little more different, than just 'marker' and 'markers' as it could get mixed up easily.

Reply:
What do you propose? I am not very keen on having variable names like `markerList` since @potay commented last time that it is unnecessary to put type in names because it is already in the definition.

Reply:
I stand by it :p Its a common style opinion on the interwebs.

Also, AFAIK, its common for people to use plural vs singular to represent a collection of things vs just one thing, but feel free to use something else if you guys prefer. 

An option would be groupOfMarkers or markerGroup, but redundant IMO. I think one instance where it might be okay to use a type in the name is for maps. Like a User ID -> name mapping could be userToNameMap. However you could also name it userNames. Preference thing.

Also, if you want to differentiate between an array and a set, you use the word unique, e.g. uniqueNames